apps/mosaico/src/Mosaico.purs

```diff
@@ -96,22 +103,35 @@ mosaicoComponent initialValues props = React.do
                          , staticPage = map StaticPageResponse props.staticPageContent
                          , tagArticlesName = props.tagArticlesName
                          , tagArticles = fold props.tagArticles
+                         , categoryStructure = props.categoryStructure
                          }
 
-  -- Listen for route changes and set state accordingly
-  useEffectOnce $ locations (routeListener setState) initialValues.nav
+  useEffectOnce do
+    Aff.launchAff_ do
+      cats <- if null props.categoryStructure
+              then Lettera.getCategoryStructure HBL
+              else pure props.categoryStructure
+      liftEffect do
+        setState _ { categoryStructure = cats }
+        -- Listen for route changes and set state accordingly
+        void $ locations (routeListener cats setState) initialValues.nav
+    pure mempty
+
   useEffect state.route do
     case state.route of
       Routes.Frontpage -> do
         if null state.frontpageArticles
         then Aff.launchAff_ do
-          frontpage <- Lettera.getFrontpage HBL
-          liftEffect $ setState \s -> s { frontpageArticles = frontpage, article = Nothing }
+          frontpage <- Lettera.getFrontpage HBL Nothing
+          liftEffect $ setState \s -> s { frontpageArticles = frontpage
```

Comment:
`setState _`

Reply:
I'm using the `s` variable in this lambda